src/vaadin-combo-box.html

```diff
@@ -397,9 +397,8 @@
           // We need to set this listener for "this.inputElement"
           // instead of just "this", otherwise keyboard navigation behaviour
           // breaks a bit on Safari and some related tests fail.
-          this.inputElement.addEventListener('keydown', e => {
+          this.addEventListener('keydown', e => {
```

Comment:
There are 2 "keydown" listeners for the `<vaadin-text-field>`, one in text-field (handles clear button) and another one here in combo-box. This one is expected to stop the propagation of the event under certain conditions so it doesn't get processed by the one in text-field (and doesn't get a dialog closed either).

combo-box is listening for the event in capture phase, but in one of the tests on Firefox, that didn't seem to matter; the text-field's listener would still get processed before this one. Moved the vaadin-combo-box' listener to the host element, so that it will for sure catch the event before it ever reaches the text-field's listener.